packages/argument-parser/src/lib/parsers/Args.ts

```diff
@@ -0,0 +1,750 @@
+import type { ChannelTypes, GuildBasedChannelTypes } from '@sapphire/discord.js-utilities';
+import { container } from '@sapphire/pieces';
+import { err, isErr, isOk, isSome, maybe, ok, type Err, type Maybe, type Ok, type Result } from '@sapphire/result';
+import type {
+	CategoryChannel,
+	DMChannel,
+	GuildMember,
+	Message,
+	NewsChannel,
+	Role,
+	StageChannel,
+	TextChannel,
+	ThreadChannel,
+	User,
+	VoiceChannel
+} from 'discord.js';
+import type * as Lexure from 'lexure';
+import { join } from 'path';
+import type { URL } from 'url';
+import { ArgumentError } from '../errors/ArgumentError';
+import { Identifiers } from '../errors/Identifiers';
+import { UserError } from '../errors/UserError';
+import type { EmojiObject } from '../resolvers/emoji';
+import type { Argument, IArgument } from '../structures/Argument';
+import { ArgumentStore } from '../structures/ArgumentStore';
+import type { Command } from '../structures/Command';
+
+container.stores.register(new ArgumentStore().registerPath(join(__dirname, '..', 'arguments')));
+
+/**
+ * The argument parser to be used in {@link Command}.
+ */
+export class Args {
+	/**
+	 * The original message that triggered the command.
+	 */
+	public readonly message: Message;
+
+	/**
+	 * The command that is being run.
+	 */
+	public readonly command: Command;
+
+	/**
+	 * The context of the command being run.
+	 */
+	public readonly commandContext: Command.RunContext;
```

Comment:
None of these should be present here, at best .run should get an extensible context which might have anything or nothing